geode-core/src/main/java/org/apache/geode/internal/cache/ClusterConfigurationLoader.java

```diff
@@ -67,18 +68,16 @@ public static void deployJarsReceivedFromClusterConfiguration(Cache cache,
     String[] jarFileNames = response.getJarNames();
     byte[][] jarBytes = response.getJars();
 
-    final JarDeployer jarDeployer = new JarDeployer(
-        ((GemFireCacheImpl) cache).getDistributedSystem().getConfig().getDeployWorkingDir());
-
     /******
      * Un-deploy the existing jars, deployed during cache creation, do not delete anything
      */
 
     if (jarFileNames != null && jarBytes != null) {
-      JarClassLoader[] jarClassLoaders = jarDeployer.deploy(jarFileNames, jarBytes);
+      List<DeployedJar> jarClassLoaders =
+          ClassPathLoader.getLatest().getJarDeployer().deploy(jarFileNames, jarBytes);
       for (int i = 0; i < jarFileNames.length; i++) {
-        if (jarClassLoaders[i] != null) {
-          logger.info("Deployed " + (jarClassLoaders[i].getFileCanonicalPath()));
+        if (jarClassLoaders.get(i) != null) {
```

Comment:
This loop is just printing the jars that got deployed from ClusterConfiguration on member startup.  If another thread deploys a jar in the middle of this loop, this list won't be updated to  include that jar.  But I believe that is the desired behavior.